In [1]:
from pathlib import Path
from keyword_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [3]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print("Full match score:", scores["perfect_match_score"])
    print("Related keywords score:", scores["related_keyword_score"])

In [4]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 954.47it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/vanilla-finetuned
Full match score: 0.4367588932806324
Related keywords score: 0.8794466403162056



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 980.05it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.50
Full match score: 0.46047430830039526
Related keywords score: 0.9071146245059288



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 972.06it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.75
Full match score: 0.45553359683794464
Related keywords score: 0.9090909090909091



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 925.71it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-1.00
Full match score: 0.45948616600790515
Related keywords score: 0.9179841897233202



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 937.44it/s, Materializing param=pooler.dense.weight]                               


models/jargon-layman-finetuned-0.25
Full match score: 0.45652173913043476
Related keywords score: 0.8992094861660079



In [5]:
print_model_scores("allenai/scibert_scivocab_uncased")

No sentence-transformers model found with name allenai/scibert_scivocab_uncased. Creating a new one with mean pooling.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1216.88it/s, Materializing param=pooler.dense.weight]                               
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  

allenai/scibert_scivocab_uncased
Full match score: 0.11462450592885376
Related keywords score: 0.29051383399209485


# Results

| Model | Exact keyword match score | Related keyword score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 0.11 | 0.29 |
| vanilla-finetuned | **0.44** | **0.88** |
| jargon-layman-finetuned | <u>0.46</u> | <u>0.92</u> |